### DISCUSSION TITLE
Connecting to Google Search Console shows empty property list.

### DISCUSSION THREAD

**Discussion by Caroga** [plausible/analytics#553] [Self-Hosted Support] [answered]
I've successfully connected to the Google Search Console as I am able to log in using both my google accounts. I have several properties under account 1 where I have just one property under account 2.
When connecting one or the other account I keep getting an empty property list.
![image](https://user-images.githubusercontent.com/4005934/103688799-ab23c600-4f92-11eb-88b6-2bc2876b5a1d.png)

Did I do something wrong? Can I enable some extra logging somwhere in order to debug this?

**Comment by metmarkosaric** [comment, marked as answer]
Thanks @Caroga! It's difficult to troubleshoot these types of questions as it doesn't depend on us. 

I can confirm that the Search Console integration works fine on self-hosted as it does on the hosted platform but what exactly is going on in your case is difficult to say. 

Please do review these instructions in detail and make sure everything is as it is explained: https://plausible.io/docs/google-search-console-integration 

And because you're self-hosting, you also need to follow these instructions in detail: https://plausible.io/docs/self-hosting-configuration#google-search-integration

That should work. Do share. And hopefully someone else who's had a similar problem can share any possible issues or details that you can look into.

**Reply by Caroga** [reply]
Thank you for your response 😄 
I've followed these instructions but will make sure to go over it again, maybe I might have missed something.
Is there any more verbose logging available that I can enable for this?

**Reply by Caroga** [reply]
Thanks to your tip I went over the documentation again and found that the following was the missing piece. 
I did not assume anymore to be done after adding the configuration and went back to the other page describing how to enable google integration, where this part is not included.
I thank you very much for your help!
![image](https://user-images.githubusercontent.com/4005934/103784013-be3da100-5039-11eb-9fe6-c46ae912eed1.png)

**Reply by Caroga** [reply]
In short, I forgot to enable the Legacy access.

**Reply by metmarkosaric** [reply]
You're welcome, glad that it helped! Thanks for the update @Caroga and enjoy Plausible!

**Reply by birkhofflee** [reply]
FYI: The Legacy API is no longer available. Enabling https://console.cloud.google.com/apis/library/searchconsole.googleapis.com is sufficient for Plausible to work (the two APIs are compatible).

**Reply by bartoszhernas** [reply]
This saved me a headache! Thanks, I missed this point from the docs as well.

**Comment by tcurdt** [comment]
> I've successfully connected to the Google Search Console as I am able to log in using both my google accounts. 

Is there somewhere a walk through in the docs that I missed?
I am somewhat lost on the Google side of things.

API Keys / OAuth 2.0 Client IDs / Service Accounts? I assume "OAuth 2.0 Client ID"?
OAuth consent screen? Which Scopes?
Which APIs? I assume just Google Search?
Authorized JavaScript origins?
Authorized redirect URIs?

**Comment by tcurdt** [comment]
I also cannot seem to find the legacy version of the "Google Search Console API". I only see this one:

https://console.developers.google.com/apis/api/searchconsole.googleapis.com/overview

**Comment by tcurdt** [comment]
After some trial and error. Some more information worth having in the docs:

    https://{foo.com}/auth/google/callback
 
is the redirect URI in question.

The list of properties is a bit strange. For me it has:

    sc-domain:foo.com
    http://foo.com
    https://foo.com

For now I have selected `https://foo.com`. I assume `sc-domain:foo.com` is the union of the whole domain?

**Reply by Vigasaurus** [reply]
Yep! I'd generally recommend using the sc-domain one, as in the past with search console alone I found lots of disparity between the individual URIs, and sc-domain batches them all together well.

**Comment by JonnieCache** [comment]
I just made this exact same mistake, missing the final step of enabling the legacy API. Suggest making it more prominent in the documentation, perhaps by numbering the steps, or at least putting it in its own paragraph.

**Reply by metmarkosaric** [reply]
thanks @JonnieCache! i've put that part as a standalone paragraph now. our docs are crowdsourced so feel free to improve things that can be made clearer. thank you!

**Comment by kees-closed** [comment]
I forgot this step: https://plausible.io/docs/self-hosting-configuration#task-three-enable-google-search-console-api
